# SNR ration computation

## 1. Load the data

In [2]:
import pandas as pd

def load_data(file_path):
    return pd.read_csv(file_path, header=None, delimiter=';')

noise_data_path = 'rest.csv'
signal_data_path = 'touch.csv'

noise_data = load_data(noise_data_path)
signal_data = load_data(signal_data_path)


Because the data comes from the MPR121 Arduino library, it is not in the 10bit format (max of 1023), so the values are very high. However, because I'm computing a ratio, it won't affect the result

## Noise values computation

I start by computing the standard deviation of the at rest signal, to establish the noise values. I use the pandas library built-in functions for that.

In [3]:
noise_std = noise_data[0].std()

print(f"Noise Standard Deviation: {noise_std}")

Noise Standard Deviation: 15.803529446059208


Then, I compute the mean of the activation signal.

In [4]:
signal_mean = signal_data[0].mean()
print(f"Signal Mean: {signal_mean}")

Signal Mean: 65449.70802919708


Once we have these value, we can compute the Signal-To-Noise Ratio using the mathematical formula :

In [5]:
import math

snr = 20 * math.log10(signal_mean / noise_std)
print(f"SNR : {snr}dB")

SNR : 72.34307247055642dB


So the SNR of the signal is around 72.343 decibels.